stl/inc/ostream

```diff
@@ -1076,6 +1082,262 @@ basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr,
     // display error code
     return _Ostr << _Errcode.category().name() << ':' << _Errcode.value();
 }
+
+#ifdef __cpp_lib_print
+#ifdef _CPPRTTI
+inline ios_base::iostate _Print_noformat_nonunicode_impl(ostream& _Ostr, const string_view _Str) {
+    // *LOCKED*
+    //
+    // This function is called from a context in which an ostream::sentry has been
+    // constructed.
+    ios_base::iostate _State = ios_base::goodbit;
+
+    _TRY_IO_BEGIN
+    const bool _Was_insertion_successful =
+        (_Ostr.rdbuf()->sputn(_Str.data(), _Str.size()) == static_cast<streamsize>(_Str.size()));
+    if (!_Was_insertion_successful) [[unlikely]] {
+        _State |= ios_base::badbit;
+    }
+    _CATCH_IO_(ios_base, _Ostr)
+
+    return _State;
+}
+
+template <bool _Add_newline>
+void _Vprint_nonunicode_impl(ostream& _Ostr, const string_view _Fmt_str, const format_args _Fmt_args) {
+    const ostream::sentry _Ok(_Ostr);
+    ios_base::iostate _State = ios_base::goodbit;
+
+    if (!_Ok) [[unlikely]] {
+        _State |= ios_base::badbit;
+    } else [[likely]] {
+        // This is intentionally kept outside of the try/catch block in _Print_noformat_nonunicode_impl()
+        // (see [ostream.formatted.print/3.2]).
+        const string _Output_str = [&]() {
+            if constexpr (_Add_newline) {
+                return _STD vformat(_Ostr.getloc(), _Fmt_str, _Fmt_args) + '\n';
+            } else {
+                return _STD vformat(_Ostr.getloc(), _Fmt_str, _Fmt_args);
+            }
+        }();
+
+        _State |= _STD _Print_noformat_nonunicode_impl(_Ostr, _Output_str);
+    }
+
+    _Ostr.setstate(_State);
+}
+
+inline ios_base::iostate _Print_noformat_unicode_impl(ostream& _Ostr, const string_view _Str) {
+    // *LOCKED*
+    //
+    // This function is called from a context in which an ostream::sentry has been
+    // constructed.
+    ios_base::iostate _State = ios_base::goodbit;
+
+    // The std::ostream& overload of vprint_unicode() expects you to determine whether the stream refers
+    // to a unicode console or not based solely on the std::ostream& provided. That class simply doesn't
+    // provide enough information to know this in every case. The best we can do (without breaking ABI)
+    // is check if the underlying std::streambuf object of the std::ostream& actually refers to a std::filebuf
+    // object. This requires the use of a dynamic_cast. (This is also why the std::ostream& overloads of
+    // std::print() et al. are deleted when RTTI is disabled.)
+    _STD streambuf* const _Streambuf = _Ostr.rdbuf();
+    _STD filebuf* const _Filebuf     = dynamic_cast<_STD filebuf*>(_Streambuf);
+
+    // If we got nullptr, then it probably isn't being used for a unicode console...
+    if (_Filebuf == nullptr) {
+        _State |= _STD _Print_noformat_nonunicode_impl(_Ostr, _Str);
+        return _State;
+    }

```

Comment:
Should we just use conditional compilation here, and always treat the stream as non-Unicode when `_CPPRTTI` is not defined (i.e. `dynamic_cast` is unavailable)?

Other concerns: we generally don't `_STD`-qualify type names, and I think it's clearer to use `auto` when the initializer is a cast expression.

Reply:
I thought about that and figured that always treating the streams as non-Unicode would be unintuitive for users disabling RTTI (i.e., the function is available, but it doesn't do what, e.g., cppreference says it should). So, I'm still personally in favor of just deleting the functions outright in that case, especially when the `FILE*` overloads still work without RTTI. (Imagine the user asking something like "Why does Cyrillic printing work when using file streams but produce garbage/mojibake with `std::cout`?.")

I'll be leaving this conversation marked as "unresolved" so that we can get further input on this matter.

Regarding the accidental `_STD`-qualifications, yeah, that was an oversight on my part. Thanks for finding that! I'll also be adding `auto` there as you mentioned.

Reply:
I'm happy with the current choice of `= delete` when RTTI is unavailable.